File: 240604/상품 코드/product-code.swift
import Foundation

precedencegroup ForwardPipe {
    associativity: left
}

infix operator |> : ForwardPipe

func |> <T, U> (value: T, function: (T) -> U) -> U {
    return function(value)
}

class Product: CustomStringConvertible {
    var name: String
    var code: Int

    init(name: String, code: Int) {
        self.name = name
        self.code = code
    }

    var description: String {
        return "product \(code) is \(name)"
    }
}

func readProduct() -> Product? {
    guard let line = readLine() else { return nil }
    let temp = line.split(separator: " ")
    guard temp.count == 2, let code = Int(temp[1]) else { return nil }
    return Product(name: String(temp[0]), code: code)
}

func main() {
    guard let product = readProduct() else { return }
    let codetree = Product(name: "codetree", code: 50)
    for i in [codetree, product] {
        print(i)
    }
}

main()

File: 250214/상품 코드/product-code.py
product_name, product_code = input().split()
product_code = int(product_code)

# Write your code here!

class product:
    def __init__(self, name = "codetree", code = 50):
        self.name = name
        self.code = code

    def redefine(self, name, code):
        self.name = name
        self.code = code
    
    def __str__(self):
        return f"product {self.code} is {self.name}"


product1 = product()
print(product1)
product1.redefine(product_name, product_code)
print(product1)